### PR TITLE
Update react-scripts and Typescript to version 4

### PR DESCRIPTION
**Pull request by ktrieu**
This PR updates react-scripts and Typescript to major version 4.

The following changes were required to deal with breaking changes:
- Uninstalling our version of `eslint` to avoid conflicts with  the version `react-scripts` provides. Our `eslintrc` has been moved to `package.json`, where [the docs](https://create-react-app.dev/docs/setting-up-your-editor/) recommend ESLint configuration is done.'
- Various changes to deal with lint errors the above changes surfaced
- Removing a [ref hack](https://github.com/UWmathNEWS/slugline-web/commit/136a6b703d9492223980a70f193ecef29900b947) around `react-router-dom` `Link`s, since the issue we were trying to work around has been fixed.
- A [change to the errors proxy](https://github.com/UWmathNEWS/slugline-web/commit/bcfdb6a61d632065bbd6b07a0341f45e090ae78e) to deal with `react-refresh` looking for `$$typeof` on every export. This should be temporary until we deliver a more lasting fix to the errors proxy.
- Setting `resetMocks` to `false` in `package.json`. CRA 4 changed this option to `true` by default which breaks `jest-mock-axios`.  Tracking issue [here](https://github.com/knee-cola/jest-mock-axios/issues/63).
- [Storing config data in JSON](https://github.com/UWmathNEWS/slugline-web/pull/85/commits/df01649b4cd31f753d2557837ee69f6e42af682c), which avoids a lint error we getting in the `config.ts` file, and makes the manifest generation code less fragile.

### CODE REVIEW

File: src/config/config.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      }
+    },
+    "manifest": {
+      "fullName": "",
+      "start_url": ".",
+      "display": "standalone",
+      "backgroundColor": "#ffffff"
+    },
+    "googleSiteVerification": ""
+  }
```
Comment (tyxchen) [inline src/config/config.json:39]: please add newline

File: src/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import config from './config.json'
+export default config
```
Comment (tyxchen) [inline src/config/index.ts:2]: please add newline

File: src/shared/components/Loader.tsx [hunk 3]
```diff
@@ -32,3 +32,5 @@ export default (props: LoaderProps) => {
   }
   return <></>;
 };
+
+export default Loader
```
Comment (tyxchen) [inline src/shared/components/Loader.tsx:36]: Please add newline and semicolon

**Review by tyxchen — CHANGES_REQUESTED** [review]
Some minor formatting stuff, then should be good to go

File: src/config/config.json (hunk 1 above)
Comment (ktrieu) [reply src/config/config.json]: https://github.com/UWmathNEWS/slugline-web/pull/85/commits/565f4fa3bd218e550b30248004596be5f6030d99

File: src/config/index.ts (hunk 2 above)
Comment (ktrieu) [reply src/config/index.ts]: https://github.com/UWmathNEWS/slugline-web/pull/85/commits/565f4fa3bd218e550b30248004596be5f6030d99

File: src/shared/components/Loader.tsx (hunk 3 above)
Comment (ktrieu) [reply src/shared/components/Loader.tsx]: https://github.com/UWmathNEWS/slugline-web/pull/85/commits/565f4fa3bd218e550b30248004596be5f6030d99

**Comment by ktrieu** [comment]
Turns out I had format on save disabled in VS Code for some reason, which explains why those format issues got through.

**Comment by tyxchen** [comment]
Awesome, it should just need a rebase on master for #86 and then we're good

**Review by tyxchen — APPROVED** [review]

**Comment by tyxchen** [comment]
So it looks like the security check got disabled because of inactivity. It should be good to merge anyways.

**Pull request merged**